### ISSUE TITLE
An error was encountered retrieving the user.

### ISSUE THREAD

**Issue by jmvizcainoio** [awlsring/terraform-provider-headscale#5]
When i try to create a .headscale_user.user i get this error from headscale. 

```
│ An error was encountered retrieving the user.
│ &{0 [] } (*models.RPCStatus) is not supported by the TextConsumer, can be resolved by supporting TextUnmarshaler interface
```

Headscale version: 0.22.3 
terraform version 1.4.6

Thanks

**Comment by awlsring** [comment]
I took a dive into this tonight to try to figure this out...

I wasn't able to reproduce the error when creating a user, but I remembered this error from the initial development and tried invalidating my authentication token. Requests with bad auth throw this error.

Digging into the code generation, I found this open issue that confirms this gap https://github.com/go-swagger/go-swagger/issues/1929. Its been open for awhile, so I don't think its gonna be addressed any time soon.

I looked into handling this error sanely, but it appears that there is another gap in the generated client is that when this is thrown, that message is always a string and can't be cast to some sort of error.

I think a larger fix I'll need to make long term if migrate from using a REST client to just using a generated gRPC client. I'm not sure how it works, but Headscale is primarily a gRPC service that as alternate access via REST, but still responds with gRPCish responses. I'm hoping migrating to gRPC for the client will give more possibilities when handling these cases.

For the short term, I'm going to commit a change the adds a function that wraps all request errors. If the error matches this string, the error returned will give a clearer signal that this is an auth issue.

TLDR: Its likely your auth token, try a new one. Let me know if that works then I'll close this issue.

**Comment by awlsring** [comment]
Closing
